### ISSUE TITLE
Variables that hold a function should have a entitiy.function scope

### ISSUE THREAD

**Issue by aziz** [bathos/Ecmascript-Sublime#4]
In all the other javascript syntaxes when you have a code like this: 

```js
var parseXML = function( data ) {}
vm.add = (taskDesc) => {} should be marked as function
var a = 12
```
`parseXML` and `add` would both gem a scope like `meta.function.js entity.name.function.js`
You can see that even github does this. I really miss it.

**Comment by bathos** [comment]
I thought this was a bug for a second, but then realized that in your example, the one you're pointing out as unscoped a property, not a variable declaration, and I hadn't thought to scope such things in this manner. It might be tougher to achieve that systematically with the grammar-focused approach of this syntax scoper than it would be in a more conventional definition, but I'll look into it.

**Comment by bathos** [comment]
The following scopes will now be applied to binding identifiers in all situations, not just within initializers:

- `entity.name.function`
- `entity.name.function.async`
- `entity.name.function.async.arrow`
- `entity.name.function.generator`
- `entity.name.function.arrow`
- `entity.name.method`
- `entity.name.method.async`
- `entity.name.method.generator`

**Comment by aziz** [comment]
It still does not work for this case: 

```js
vm.setSelectedStop = function(stop) {}
```
<img width="385" alt="screen shot 2016-01-01 at 2 01 58 pm" src="https://cloud.githubusercontent.com/assets/3202/12070898/482aafe2-b090-11e5-8ae7-c958485eddf1.png">

**Comment by bathos** [comment]
@aziz There was a window in there where I put up a release (0.2.6) before I caught that oversight (0.2.6.1, a few minutes later). PackageControl noticed the first one immediately but there's a minimum wait before it will pick up the next change. Right now, PackageControl is actually down:

![image](https://cloud.githubusercontent.com/assets/6257356/12072042/a637c228-b09a-11e5-8ef6-636ead5433e5.png)

...and I don't know how long that's been true, so I'm not sure if they captured the 0.2.6.1 update before the site died.

**Comment by bathos** [comment]
I tested this with your theme locally and can confirm that your last example works with the latest updates. I also tested some other cases with your theme though and found that it was still missing cases where the function was an initializer value, despite those scopes having always existed. The reason for that is that other scopes were taking precedence over it.

I've adjusted those scopes now (will release in a few minutes) to share the same selector 'weight' as the newly introduced ones. I also made them less specific: previously, in initializers, it would only consider the variable to be the 'name' if the function itself was anonymous.

For some background on all this: I had done it the way it was originally on purpose. Personally, I prefer very grammar-y highlighting, and felt uncomfortable with the consequences of this idea...

```
const derp = () => {};
const larp = derp;
```

That `derp` would receive a special scope and larp would not, despite the fact that from a grammar perspective, there is nothing at all different about them: they are both just binding identifiers in a const declaration, and even semantically they are both being initialized to have functions as their value. But I figure you're right, people are used to this behavior and expect it, and it's not a great loss from my perspective.
